test/mochitest/browser_dbg-xhr-breakpoints.js

```diff
@@ -28,37 +35,46 @@ async function removeXHRBreakpoint(dbg, index) {
 }
 
 function getXHRBreakpointsElements(dbg) {
-  return [...dbg.win.document.querySelectorAll(".xhr-breakpoints-pane .xhr-container")];
+  return [
+    ...dbg.win.document.querySelectorAll(".xhr-breakpoints-pane .xhr-container")
+  ];
 }
 
 function getXHRBreakpointLabels(elements) {
   return elements.map(element => element.title);
 }
 
 function getXHRBreakpointCheckbox(dbg) {
-  return findElementWithSelector(dbg, ".xhr-breakpoints-pane .breakpoints-exceptions input");
+  return findElementWithSelector(
+    dbg,
+    ".xhr-breakpoints-pane .breakpoints-exceptions input"
+  );
 }
 
 async function clickPauseOnAny(dbg, expectedEvent) {
   getXHRBreakpointCheckbox(dbg).click();
   await waitForDispatch(dbg, expectedEvent);
 }
 
-// Tests that a basic XHR breakpoint works for get and POST is ignored
 add_task(async function() {
   const dbg = await initDebugger("doc-xhr.html", "fetch.js");
-  await dbg.actions.setXHRBreakpoint("doc", "GET");
+  const wait = ms => new Promise(resolve => setTimeout(resolve, ms));
+
+  await addXHRBreakpoint(dbg, "doc", "GET");
+
   invokeInTab("main", "doc-xhr.html");
   await waitForPaused(dbg);
   assertPausedLocation(dbg);
-  resume(dbg);
+  await resume(dbg);
 
   await dbg.actions.removeXHRBreakpoint(0);
   invokeInTab("main", "doc-xhr.html");
+  await wait(1000); // timeout needed before checking whether it's paused
   assertNotPaused(dbg);
 
-  await dbg.actions.setXHRBreakpoint("doc-xhr.html", "POST");
-  invokeInTab("main", "doc");
+  await addXHRBreakpoint(dbg, "doc", "POST");
+  invokeInTab("main", "doc-xhr.html");
+  await wait(1000); // timeout needed before checking whether it's paused
```

Comment:
Oops, we'll want a way around these waits; did you try my suggestion re: polling instead of a generic number?